src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -1174,7 +1168,11 @@ private bool ScanNumericLiteral(ref TokenInfo info)
             {
                 this.AddError(MakeError(start, TextWindow.Position - start, ErrorCode.ERR_InvalidNumber));
             }
-            if (usedUnderscore)
+            else if (firstCharWasUnderscore)
+            {
+                CheckFeatureAvailability(MessageID.IDS_FeatureLeadingDigitSeparator);
```

Comment:
Consider moving the new CheckFeatureAvailability call into the hex/binary condition.
Or, if this call can affect other scenarios than hex/binary, then let's ensure those code paths have test coverage too. Thanks

Reply:
The we would still need to check `!firstCharWasUnderscore` to not get two feature availability errors for underscore and leading underscore.